### PR TITLE
Updates for .NET May 2024 Servicing

### PR DESCRIPTION
**Pull request by lbussell**

### CODE REVIEW

File: tests/Microsoft.DotNet.Docker.Tests/TestData.cs [hunk 1]
```diff
@@ -202,22 +245,29 @@ public static class TestData
...
-                    ImageVariant = DotNetImageVariant.Composite | DotNetImageVariant.Extra, SupportedImageRepos = DotNetImageRepo.Aspnet },
-            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Arm,     SdkOS = OS.Jammy,
-                    ImageVariant = DotNetImageVariant.Extra, SupportedImageRepos = DotNetImageRepo.Runtime_Deps | DotNetImageRepo.Runtime | DotNetImageRepo.Aspnet },
-            new ProductImageData { Version = V9_0, OS = OS.JammyChiseled,       Arch = Arch.Arm,     SdkOS = OS.Jammy, SdkImageVariant = DotNetImageVariant.AOT,
+            new ProductImageData { Version = V9_0, OS = OS.Alpine319,           Arch = Arch.Arm,     SdkOS = OS.Alpine319,
+                    ImageVariant = DotNetImageVariant.Extra, SupportedImageRepos = DotNetImageRepo.Runtime_Deps },
+            new ProductImageData { Version = V9_0, OS = OS.Alpine319,           Arch = Arch.Arm,     SdkOS = OS.Alpine319, SdkImageVariant = DotNetImageVariant.AOT,
                     ImageVariant = DotNetImageVariant.AOT, SupportedImageRepos = DotNetImageRepo.Runtime_Deps }
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/TestData.cs:254]:
```suggestion
                    ImageVariant = DotNetImageVariant.AOT, SupportedImageRepos = DotNetImageRepo.Runtime_Deps },
```

File: manifest.json [hunk 2]
```diff
@@ -1961,29 +1961,29 @@
         {
           "productVersion": "$(dotnet|9.0|product-version)",
           "sharedTags": {
-            "$(dotnet|9.0|fixed-tag)-cbl-mariner2.0": {},
-            "$(dotnet|9.0|floating-tag)-cbl-mariner2.0": {}
+            "$(dotnet|9.0|fixed-tag)-azurelinux3.0": {},
```
Comment (mthalman) [inline manifest.json:2067]: We can't include any 9.0 changes in this PR because that's not being released tomorrow. Including them here would cause these changes to be published tomorrow. You should just be able to revert entire 42574eabc89d8344169933d509cefc782cca2e7c commit.

File: README.aspnet.md [hunk 3]
```diff
@@ -113,11 +118,11 @@ Tags | Dockerfile | OS Version
...
-9.0.0-preview.3-jammy-chiseled-extra-amd64, 9.0-preview-jammy-chiseled-extra-amd64, 9.0.0-preview.3-jammy-chiseled-extra, 9.0-preview-jammy-chiseled-extra | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/jammy-chiseled-extra/amd64/Dockerfile) | Ubuntu 22.04
-9.0.0-preview.3-jammy-chiseled-composite-amd64, 9.0-preview-jammy-chiseled-composite-amd64, 9.0.0-preview.3-jammy-chiseled-composite, 9.0-preview-jammy-chiseled-composite | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/jammy-chiseled-composite/amd64/Dockerfile) | Ubuntu 22.04
-9.0.0-preview.3-jammy-chiseled-composite-extra-amd64, 9.0-preview-jammy-chiseled-composite-extra-amd64, 9.0.0-preview.3-jammy-chiseled-composite-extra, 9.0-preview-jammy-chiseled-composite-extra | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/jammy-chiseled-composite-extra/amd64/Dockerfile) | Ubuntu 22.04
+9.0.0-preview.3-noble-amd64, 9.0-preview-noble-amd64, 9.0.0-preview.3-noble, 9.0-preview-noble | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/noble/amd64/Dockerfile) | Ubuntu 24.04
```
Comment (mthalman) [inline README.aspnet.md:121]: Again, 9.0 changes shouldn't be included in this PR.

File: src/aspire-dashboard/8.0/cbl-mariner-distroless/amd64/Dockerfile [hunk 4]
```diff
@@ -11,8 +11,6 @@ RUN tdnf install -y \
 # Retrieve Aspire Dashboard
 RUN dotnet_aspire_version=8.0.0-preview.7.24251.11 \
     && curl -fSL --output aspire_dashboard.zip https://dotnetbuilds.azureedge.net/public/aspire/$dotnet_aspire_version/aspire-dashboard-linux-x64.zip \
-    && aspire_dashboard_sha512='8c923e139d28baba92901456fee1bfbe31f037a405d6754ed6acadaf8775399be4d6afb33bf9d5d374e2966e3227fd3faa7519e7aceee84bb65652d8ca13ad4b' \
-    && echo "$aspire_dashboard_sha512  aspire_dashboard.zip" | sha512sum -c - \
```
Comment (mthalman) [inline src/aspire-dashboard/8.0/cbl-mariner-distroless/amd64/Dockerfile:15]: These shouldn't be removed.

File: README.aspnet.md [hunk 5]
```diff
@@ -222,6 +222,8 @@ Tags | Dockerfile | OS Version
 Tags | Dockerfile | OS Version
 -----------| -------------| -------------
 9.0.0-preview.3-bookworm-slim-arm32v7, 9.0-preview-bookworm-slim-arm32v7, 9.0.0-preview.3-bookworm-slim, 9.0-preview-bookworm-slim, 9.0.0-preview.3, 9.0-preview | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/bookworm-slim/arm32v7/Dockerfile) | Debian 12
+9.0.0-preview.3-alpine3.19-arm32v7, 9.0-preview-alpine3.19-arm32v7, 9.0-preview-alpine-arm32v7, 9.0.0-preview.3-alpine3.19, 9.0-preview-alpine3.19, 9.0-preview-alpine | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/alpine3.19/arm32v7/Dockerfile) | Alpine 3.19
+9.0.0-preview.3-alpine3.19-composite-arm32v7, 9.0-preview-alpine3.19-composite-arm32v7, 9.0-preview-alpine-composite-arm32v7, 9.0.0-preview.3-alpine3.19-composite, 9.0-preview-alpine3.19-composite, 9.0-preview-alpine-composite | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/main/src/aspnet/9.0/alpine3.19-composite/arm32v7/Dockerfile) | Alpine 3.19
```
Comment (mthalman) [inline README.aspnet.md:226]: These shouldn't be added until next week. This should just be a revert of the entire 901f62cc9ce01c173a221d76285cddfca772e14d commit.

**Review by mthalman — COMMENTED** [review]
I left specific comments about this but, in general, make sure to check the diff of your changes and not include any .NET 9 Dockerfile changes since that isn't shipping tomorrow.

File: README.aspnet.md (hunk 3 above)
Comment (lbussell) [reply README.aspnet.md]: Resolved all of these by removing several commits from this PR.

File: manifest.versions.json [hunk 6]
```diff
@@ -119,26 +120,27 @@
...
+    "mingit|x64|url": "https://github.com/git-for-windows/git/releases/download/v2.45.0.windows.1/MinGit-2.45.0-64-bit.zip",
+    "mingit|x64|sha": "f607bbd459bae73369e6509fe849c4c48152f4d33b0021d5881e1e9e7ae79e26",
 
-    "monitor|6.3|build-version": "6.3.5",
-    "monitor|6.3|linux-musl|x64|sha": "fdfcac194efd44e62ef642d5a82c568a0b2d4b08f627263e9cab4a5586c7b20713307a57a7547e7a659824bb935e90eadd0dedc96b143975148b046ea8f44a00",
-    "monitor|6.3|linux-musl|arm64|sha": "3590323ab4b30093d73ca5d9459988908b1952a67e0b4dd228a37731a93b164b91734f7d3dc47e7e30232e30e3d4b36f545533fe64fee93bc413bd22c1dc4fb0",
-    "monitor|6.3|linux|x64|sha": "4834f0f2a0d0037b3c3cbb4cc25875a3da175b54866ba7c731c1e967c3dfdde40def8ac434b71bbf8e591ed69df6bc89c9772cf688fa444ea1daf655bfd2a5e3",
-    "monitor|6.3|linux|arm64|sha": "f33f22f1d47155d738fc379b85537d6da096b319d4fbbd58badbd7286e07124078c15d410cfb2082027741f3b9c65d7ffdb0e65088f5bfc7aa8f4429351361f2",
-    "monitor|6.3|product-version": "6.3.5",
+    "monitor|6.3|build-version": "6.3.6-servicing.24258.3",
```
Comment (jander-msft) [inline manifest.versions.json:128]: Please update .NET Monitor to the final versions:
- .\eng\Set-DotnetVersions.ps1 6.3 -MonitorVersion 6.3.6
- .\eng\Set-DotnetVersions.ps1 7.3 -MonitorVersion 7.3.4
- .\eng\Set-DotnetVersions.ps1 8.0 -MonitorVersion 8.0.2

File: src/sdk/9.0/jammy-aot/amd64/Dockerfile
Comment (mthalman) [inline src/sdk/9.0/jammy-aot/amd64/Dockerfile:1]: These jammy-aot files shouldn't be included.

File: manifest.versions.json [hunk 7]
```diff
@@ -119,26 +120,27 @@
     "libssl|buster": "1.1",
     "libssl|focal": "1.1",
     "libssl|jammy": "3",
+    "libssl|noble": "3t64",
 
-    "mingit|x64|url": "https://github.com/git-for-windows/git/releases/download/v2.44.0.windows.1/MinGit-2.44.0-64-bit.zip",
-    "mingit|x64|sha": "ed4e74e171c59c9c9d418743c7109aa595e0cc0d1c80cac574d69ed5e571ae59",
+    "mingit|x64|url": "https://github.com/git-for-windows/git/releases/download/v2.45.0.windows.1/MinGit-2.45.0-64-bit.zip",
+    "mingit|x64|sha": "f607bbd459bae73369e6509fe849c4c48152f4d33b0021d5881e1e9e7ae79e26",
```
Comment (mthalman) [inline manifest.versions.json:126]: This is causing changes to the 9.0 SDK Dockerfiles. This should probably follow the same pattern as rocks-toolbox so that we can control which mingit version is used for each .NET version.

Comment (lbussell) [reply manifest.versions.json]: Fixed in https://github.com/dotnet/dotnet-docker/pull/5460/commits/3709994190fe49f45b8b1b4db2c025471d240b0b

**Review by jander-msft — APPROVED** [review]
.NET Monitor LGTM

**Review by mthalman — APPROVED** [review]
